spec.html

```diff
@@ -42955,6 +42966,383 @@ <h1>Properties of _TypedArray_ Instances</h1>
+                1. Let _error_ be a new *SyntaxError* exception.
+                1. Return the Record { [[Read]]: _read_, [[Bytes]]: _bytes_, [[Error]]: _error_ }.
+              1. Else if _char_ is *"-"*, then
+                1. Set _char_ to *"+"*.
+              1. Else if _char_ is *"_"*, then
+                1. Set _char_ to *"/"*.
+            1. If the sole code unit of _char_ is not an element of the standard base64 alphabet, then
+              1. Let _error_ be a new *SyntaxError* exception.
+              1. Return the Record { [[Read]]: _read_, [[Bytes]]: _bytes_, [[Error]]: _error_ }.
+            1. Let _remaining_ be _maxLength_ - the length of _bytes_.
+            1. If _remaining_ = 1 and _chunkLength_ = 2, or if _remaining_ = 2 and _chunkLength_ = 3, then
+              1. Return the Record { [[Read]]: _read_, [[Bytes]]: _bytes_, [[Error]]: ~none~ }.
+            1. Set _chunk_ to the string-concatenation of _chunk_ and _char_.
+            1. Set _chunkLength_ to the length of _chunk_.
+            1. If _chunkLength_ = 4, then
+              1. Set _bytes_ to the list-concatenation of _bytes_ and ! DecodeBase64Chunk(_chunk_).
+              1. Set _chunk_ to the empty String.
+              1. Set _chunkLength_ to 0.
+              1. Set _read_ to _index_.
+              1. If the length of _bytes_ = _maxLength_, then
+                1. Return the Record { [[Read]]: _read_, [[Bytes]]: _bytes_, [[Error]]: ~none~ }.
+        </emu-alg>
+      </emu-clause>
+
+      <emu-clause id="sec-fromhex" type="abstract operation">
+        <h1>
+          FromHex (
+            _string_: a string,
+            optional _maxLength_: a non-negative integer,
+          ): a Record with fields [[Read]] (an integral Number), [[Bytes]] (a List of byte values), and [[Error]] (either ~none~ or a *SyntaxError* object)
+        </h1>
+        <dl class="header">
+        </dl>
+        <emu-alg>
+          1. If _maxLength_ is not present, let _maxLength_ be 2<sup>53</sup> - 1.
+          1. Let _length_ be the length of _string_.
+          1. Let _bytes_ be « ».
+          1. Let _read_ be 0.
+          1. If _length_ modulo 2 is not 0, then
+            1. Let _error_ be a new *SyntaxError* exception.
+            1. Return the Record { [[Read]]: _read_, [[Bytes]]: _bytes_, [[Error]]: _error_ }.
+          1. Repeat, while _read_ &lt; _length_ and the length of _bytes_ &lt; _maxLength_,
+            1. Let _hexits_ be the substring of _string_ from _read_ to _read_ + 2.
+            1. If _hexits_ contains any code units which are not in *"0123456789abcdefABCDEF"*, then
+              1. Let _error_ be a new *SyntaxError* exception.
+              1. Return the Record { [[Read]]: _read_, [[Bytes]]: _bytes_, [[Error]]: _error_ }.
+            1. Set _read_ to _read_ + 2.
+            1. Let _byte_ be the integer value represented by _hexits_ in base-16 notation, using the letters A-F and a-f for digits with values 10 through 15.
```

Comment:
```suggestion
            1. Let _byte_ be the integer value represented by _hexits_ in base-16 notation using the letters *A* through *F* and *a* through *f* for the hexits with values 10 through 15.
```

This is a *bit* hand-wavey. There was no easy way to re-use MV or something?

Reply:
this suggestion will necessitate defining the term "hexit" somewhere, fwiw.

Reply:
I don't think so. Like "digit" and "bit", "hexit" is a widely-used term in our field: https://en.wiktionary.org/wiki/hexit

Reply:
fwiw I've never encountered it before this PR, despite being quite familiar with hexadecimal numbers and notation in multiple contexts.